### PR TITLE
[NEW] dummy auth model

### LINKED ISSUE #156: [TODO]  Create a simple-local-auth component to improve out-of-box DevEx
Create a compatible (no changes to current auth API)   auth component that uses local DB and only has userid field.

Keep this component as simple as possible ( it will never be used in production, only in dev and testing).

### PR DESCRIPTION
**Pull request by sidmohanty11**
This PR introduces,
- [x] a dummy authentication model
- [x] removes firebase error
- [x] fixes all the console errors for a smooth development experience

Fixes #156

### CODE REVIEW

**Comment by sidmohanty11** [comment]
## Dummy User Authentication Model

https://user-images.githubusercontent.com/73601258/174348400-90d4baa5-4026-4e1f-9fe5-8851cddff741.mp4

**Comment by sidmohanty11** [comment]
I've fixed almost all errors that were arising:
- firebase error
- `slideCount` error

There remains only one error (`<button> inside <button>`) which is coming from the Menubar component because we are using Navbar.Toggle but inside for styling, we are using another button. This should be fixed with the new Menubar component which is being worked on in this [PR](https://github.com/RocketChat/RC4Community/pull/155).

image

File: app/lib/auth/RocketChatOAuthProvider.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ export function RocketChatOAuthProvider(options = {}) {
     type: "oauth",
     authorization: urlJoin(options.rocketChatUrl,'/oauth/authorize'),
     scope: "openid email profile",
-    token: urlJoin(process.env.NEXTAUTH_URL,'/api/auth/rocketchat-token-legacy'),
+    token: urlJoin(process.env.NEXTAUTH_URL ?? "http://localhost:3000",'/api/auth/rocketchat-token-legacy'),
```
Comment (debdutdeb) [inline app/lib/auth/RocketChatOAuthProvider.js:9]:
```suggestion
    token: urlJoin(process.env.NEXTAUTH_URL || "http://localhost:3000",'/api/auth/rocketchat-token-legacy'),
```

File: app/pages/api/auth/[...nextauth].js [hunk 2]
```diff
@@ -20,9 +20,9 @@ export default async function handleAuth(req,res){
                 }
             }),
 			RocketChatOAuthProvider({
-				clientId: process.env.ROCKETCHAT_CLIENT_ID,
-				clientSecret: process.env.ROCKETCHAT_CLIENT_SECRET,
-				rocketChatUrl: process.env.ROCKETCHAT_URL
+				clientId: process.env.ROCKETCHAT_CLIENT_ID ?? "random",
+				clientSecret: process.env.ROCKETCHAT_CLIENT_SECRET ?? "random",
+				rocketChatUrl: process.env.ROCKETCHAT_URL ?? "http://localhost:3000"
```
Comment (debdutdeb) [inline app/pages/api/auth/[...nextauth].js:25]: Make sure empty strings aren't an issue here.

File: app/components/auth/dummy/ui/DummyLoginButton.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from "react";
+import { NoUserAvatar } from "../../NoUserAvatar";
+import DummyLoginUI from "./DummyLoginUI";
+import styles from "../styles/DummyLoginButton.module.css";
+
+export default function DummyLoginButton() {
+  const [isOpen, setOpen] = useState(false);
```
Comment (debdutdeb) [inline app/components/auth/dummy/ui/DummyLoginButton.js:7]: Use more explicit names. What is `open`ness supposed to represent here? suggestion: `isLoginUiOpen` never be afraid of over-explained variable names, most of us love them.

File: app/components/auth/dummy/lib/function.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      "https://user-images.githubusercontent.com/25859075/29918905-88dcc646-8e5c-11e7-81ec-242bc58dce1b.jpg",
+    email: "dummyuser@rocket.chat",
+    emailVerified: false,
+    phoneNumber: null,
+    displayName: "dummy.cat",
+  };
+};
+
+export const dummyUserLogin = () => {
+  let user = JSON.parse(sessionStorage.getItem("dummy_user"));
```
Comment (debdutdeb) [inline app/components/auth/dummy/lib/function.js:15]: No need for this, you're already checking session storage on `DummyLoginButton`.

File: app/components/auth/dummy/ui/DummyLoginUI.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+export default function DummyLoginUI({ user, setUser }) {
+  const login = () => {
+    const dummy_user = dummyUserLogin();
+    setUser(dummy_user);
+    sessionStorage.setItem("dummy_user", JSON.stringify(dummy_user));
+  }
+  const logout = () => {
+    setUser({});
+    sessionStorage.removeItem("dummy_user");
+  }
```
Comment (debdutdeb) [inline app/components/auth/dummy/ui/DummyLoginUI.js:14]: Move them to `DummyLoginButton` and just pass the `user` state. Pass these as `handleLogin` and `handleLogout`.

File: app/components/auth/dummy/ui/DummyLoginUI.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  return (
+      !user.id ? <div className="d-flex flex-column align-items-center my-3">
```
Comment (debdutdeb) [inline app/components/auth/dummy/ui/DummyLoginUI.js:16]: Shift the login please, `user.id ? ... : ...`

File: app/components/menubar.js [hunk 7]
```diff
@@ -73,33 +81,45 @@ export default function Menubar(props) {
...
-        <Dropdown.Menu size="sm" title="">
-          <Dropdown.Header>RC4Community Profile</Dropdown.Header>
-          <Dropdown.Item><Link href={`/profile/${Cookies.get('user')}`}><a className={styles.dropdown_menu_item}>Profile</a></Link></Dropdown.Item>
-        </Dropdown.Menu>
-      </Dropdown>
-      :
-      ""
-      }
-      </div>
+          {Cookies.get("user") ? (
```
Comment (debdutdeb) [inline app/components/menubar.js:101]: Save the cookie value in some variable pls.

File: app/components/menubar.js [hunk 8]
```diff
@@ -73,33 +81,45 @@ export default function Menubar(props) {
...
+            <Dropdown align="end" className={styles.dropdown_menu}>
+              <Dropdown.Toggle as={CustomToggle} />
+              <Dropdown.Menu size="sm" title="">
+                <Dropdown.Header>RC4Community Profile</Dropdown.Header>
+                <Dropdown.Item>
+                  <Link href={`/profile/${Cookies.get("user")}`}>
```
Comment (debdutdeb) [inline app/components/menubar.js:107]: Variable for the cookie.

**Review by debdutdeb — CHANGES_REQUESTED** [review]

**Review by Sing-Li — CHANGES_REQUESTED** [review]
image


Works well @sidmohanty11      But what happened to the menu/TopNav?    It would be good to have the menu change depending on the logged in state as well.

**Comment by sidmohanty11** [comment]
It is there in my case.
image

Is it due to cms uninitialized? But that should break the app, right? Can you check with running cms?

File: app/components/auth/dummy/ui/DummyLoginButton.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+export default function DummyLoginButton() {
+  const [isLoginUiOpen, setIsLoginUiOpen] = useState(false);
+  const [user, setUser] = useState({});
+
+  useEffect(() => {
+      const isStoredInSession = JSON.parse(sessionStorage.getItem("dummy_user"));
+      if (isStoredInSession) {
+          setUser(isStoredInSession);
+      }
+  },[])
```
[the remark this answers is not in the document]
Comment (debdutdeb) [reply app/components/auth/dummy/ui/DummyLoginButton.js:15]: Move all `createDummyUser` or validation functions in the same file.

[the remark this answers is not in the document]
Comment (sidmohanty11) [reply app/components/auth/dummy/ui/DummyLoginButton.js:15]: Great idea! Ah, I just went with the conventional code structure of other auths. I will make the changes!

**Comment by sidmohanty11** [comment]
Did a clean clone and install, the top nav works for me. 
image

**Pull request merged**